### PR TITLE
FAQ, Getting Started, and Improve Schema Docs

### LINKED ISSUE #106: add primitive type to neurodata type RTD specs
The schema specification on Read the Docs should display the primitive type for each specification. For example, [ElectrodeTableRegion](http://nwb-schema.readthedocs.io/en/latest/format.html#sec-electrodetableregion) should explicitly call out that it is a Dataset, and [ElectrodeGroup](http://nwb-schema.readthedocs.io/en/latest/format.html#sec-electrodegroup) should explicitly call out that it is a Group.

### PR DESCRIPTION
**Pull request by oruebel**
**General docs**
* Added FAQ page
* Updated resources page to a getting started page
* Updated ordering of cos in the main index

**Format schema docs**
* Updated the source generator to improve the layout of the description of objects by summarizing all properites in a single list, rather than having additional items listed at the beginning
* Added listing of the primitive type to Fix #106

### CODE REVIEW

File: docs/general/source/faq.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+See the :ref:`sec_getting_started` section for more information.
+
+What is the difference between pynwb, nwb-schema and api-python?
+----------------------------------------------------------------
+
+``PyNWB`` is the current Python reference read/write API for the NWB:N 2.x format.
+The ``nwb-schema`` repo is used to manage development on the format schema. End-users
+who want to use NWB:N typically do not need to worry about the ``nwb-schema`` repo
+as the current schema is always installed with the correspondign API (whether it
```
Comment (jcfr) [inline docs/general/source/faq.rst:16]: correspondign -> corresponding

File: docs/general/source/faq.rst [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+is PyNWB for Python or MatNWB for Matlab). ``api-python`` is a deprecated write-only
+API designed for NWB:N 1.0.x files.
```
Comment (jcfr) [inline docs/general/source/faq.rst:18]: Should a pointer to `pynwb/legacy` be added here ?

File: docs/general/source/faq.rst [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+---------------------------------------------------
+
+Yes
+
+Does it make sense to start using NWB:N 2.0 now, or is it more sensible to wait?
+--------------------------------------------------------------------------------
+
+Yes. A first public beta release of PyNWB (for Python 2.7.x and >3.5) and NWB 2.0 has been released in
+November 2017 in conjunction with the Annual Society for Neuroscience Meeting (SfN). The intent of this beta
+release is to enable early adopters to start exploring the new format and software. While development on NWB 2.0 has
```
Comment (jcfr) [inline docs/general/source/faq.rst:40]: is to enable early adopters -> was to enable early adopters

File: docs/general/source/getting_started.rst [hunk 4]
```diff
@@ -1,6 +1,28 @@
...
+Getting Started
+***************
+
+How to get started
+------------------
+
+I am scientist/developer in a neuroscience lab and want to start using NWB:N
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For end-users, the main entry point into the world of NWB:N are usually the PyNWB Python API.
```
Comment (jcfr) [inline docs/general/source/getting_started.rst:13]: are usually -> is usually

**Comment by oruebel** [comment]
@jcfr I guess I was a bit fast in just merging these changes. I'll make the edits you suggested and create another pull request.

**Pull request merged**